module2/lessons/js_event_listeners.md

```diff
@@ -0,0 +1,297 @@
+
+
+## Vocabulary
+
+- `Event` Any event which takes place in the DOM, these can be generated by users or the browser.
+- `Event Listener` A function invoked on a DOM node which fires the `event handler` when a specified event occurs on the node or its children
+- `Event Handler` A function that will run when a specific event occurs
+- `event.target` The element that is the "target" of the event. For example, the element that is clicked on in a click event.
+
+<section class="call-to-action">
+### Warm Up
+Complete the tasks listed in the JS file on [this codepen](https://codepen.io/turingschool/pen/MWMeopJ?editors=1010).
+</section>
+
+## Event Listeners
+
+Changing stuff on the page with JavaScript is great, but you might as well have just written it in the markup to begin with. The real power of JavaScript comes into play when we write code that **responds to user input**.
+
+This power emerges when we start **listening for user events**. This is the crux of front-end engineering. We present a user interface and then as the user interacts with the UI, we change and update what they see.
+
+<section class="call-to-action">
+Let's revisit the codepen from the Warm Up and add in an Event Listener. We still want to change the title, fact, and image, but we want to do so only when a users clicks a button.  
+  
+- Add a "Click Me" button to your html above the <h1> element. Let's give it an id of 'click-me'.
+- In your js file, use `document.querySelector()` to grab that button by its id and save it to a variable
+- Now add an eventListener to the button.
+  - We want the eventListener to listen for a "click" event that happens on that button. 
+  - When the "click" happens on the button, we want to invoke a function that does all the things we want to do. Let's call that function changePageContent().
+- In the warmup, you wrote code to change the title, fact and image.  Move this code into a changePageContent() function that will be invoked by the eventListener.
+
+Consider:  
+- Why don't we invoke the function within the eventListener?  What happens if we do?
+</section>
+
+## Explore
+
+There are many types of events beyond just "click".   Explore this list of [Mouse Events](https://developer.mozilla.org/en-US/docs/Web/API/Element#mouse_events) on MDN. Scroll beyond just mouse events to explore keyboard events, focus events, etc.  
+
+Open up a webpage that you often visit. Maybe it's a social media page, maybe it's your bank's website, maybe it's a news site. Take a few minutes and list all the event listeners that you think might be on that page! What do they do?
+
+<section class="call-to-action">
+### Pair Challenge
+
+DOM manipulation based on user interaction requires putting several pieces together which is challenging! It is valuable to pseudocode out the steps of what you're trying to do before you try to do it in code.
+
+Fork [this Codepen](https://codepen.io/turingschool/pen/qBzNraG). 
```

Comment:
We could consider adding a "Switch driver/navigator positions" halfway through this activity to give both students practice

Reply:
I set up to have them switching roles for the next breakout room practice so they each get a turn in the driver seat :)